digits/static/js/table_selection.js

```diff
@@ -0,0 +1,210 @@
+            var last = Math.max(that.mousedown_row.rowIndex, this.rowIndex) + 1;
+
+            // Do we want to add, remove, or replace the selection
+            var action = (that.mousedown_with_shift ? 'add' :
+                          that.mousedown_with_alt ? (
+                              that.mousedown_was_selected ? 'remove' : 'add') :
+                          'replace');
+
+            if (action == 'remove') {
+                for (var i = first; i <= last; i++)
+                    deselect($(table_id + ' tr').eq(i));
+                that.last_selected_row = $(TableSelection.current_table + ' tr.selected').last();
+            } else {
+                if (action == 'replace')
+                    deselect($(table_id + ' tr'))
+
+                for (var i = first; i <= last; i++)
+                    select($(table_id + ' tr').eq(i));
+
+                that.last_selected_row = this;
+            }
+            that.mousedown_last_rowIndex = this.rowIndex;
+        }
+    });
+
+    // end drag selection
+    $(table_id + ' tbody').on('mouseup', 'tr', function() {
+        that.mousedown_row = null;
+    });
+
+    // Let links continue to work.  Without this, a click would
+    // only do a row selection.
+    $(table_id + ' tbody').on('click', 'a', function(e) {
+        location.href = $(this).attr('href');
+    });
+
+    // click selection
+    $(table_id + ' tbody').on('click', 'tr', function(e) {
+        e.preventDefault();
+        // shift select region
+        if ((e.shiftKey)) {
+            if (that.last_selected_row != null) {
+                var first = Math.min(that.last_selected_row.rowIndex, this.rowIndex) + 1;
+                var last = Math.max(that.last_selected_row.rowIndex, this.rowIndex) + 1;
+                for (var i = first; i <= last; i++) {
+                    select($(table_id + ' tr').eq(i));
+                }
+                that.last_selected_row = this;
+            }
+        }
+        // toggle click selection
+        else if ((e.ctrlKey || e.metaKey)) {
+            if ($(this).hasClass('selected')) {
+                deselect($(this));
+                that.last_selected_row = $(TableSelection.current_table + ' tr.selected').last();
+            } else {
+                select($(this));
+                that.last_selected_row = this;
+            }
+        }
+        // simple unmodified click selection
+        else {
+            deselect($(table_id + ' tr'));
+            select($(this));
+            that.last_selected_row = this;
+        }
+    });
+
+    // Because the key event is global, only register once.
+    if (!TableSelection.initialized) {
+        TableSelection.initialized = true;
+        $(document).keydown(function(e) {
+            if (TableSelection.current_table == null) return;
+            // Use alt-a for select all rows
+            if ((e.ctrlKey || e.metaKey) && e.keyCode == 65) {
```

Comment:
Funny that you mention Chrome and Firefox, because I'd been developing using Safari, and when I went to test on Chrome and Firefox, several things stopped working, because Safari still supported deprecated javascript.

This wasn't from that though. I"d tried to minimize the selection_changed events, and had a fault  in the logic.  Thanks for catching it.  I've tested the fix on Safari, Firefox, and Chrome.